### PR TITLE
🔀➕ Flyway Integration

### PR DESCRIPTION
**Pull request by Tuckerle**
**Description:**  

Flyway as database migration tool integrated.

**Reference:**

Issue: #64 

Notifying additional team members:

@mirrodi @MrSebastian @AnHo314

---
**Definition of Done (DoD):**

- [ ] Unittests gepflegt
- [ ] Dokumentation ist gepflegt

### CODE REVIEW

File: doc/src/documentation/guides/database-migration/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Database Migration
+
+Our database migrations are handled by flyway.
+This way we can version our DDL files.
+They are automatically applied when the backend starts and there are new, not yet applied, migrations.
+
+For this to work if you have to make changes to the database you have to create a new SQL script in the migration folder.
+The migrations folder is located in the backend `src/main/resources/db/migration`.
+The scripts name has to follow this convention `V<version_number>__<description_split_with_underscores>`.
```
Comment (LuisEinig) [inline doc/src/documentation/guides/database-migration/README.md:9]: Should this be `2` underscores? The example from the next line only uses `1`

File: praktikumsplaner-backend/src/main/resources/application-dev.yml [hunk 2]
```diff
@@ -13,13 +13,7 @@ spring:
   h2.console.enabled: true
   jpa:
     database: H2
-    hibernate:
-        # always drop and create the db should be the best
-        # configuration for local (development) mode. this
-        # is also the default, that spring offers by convention.
-        # but here explicite:
-        ddl-auto: create-drop
-        naming.physical-strategy: org.hibernate.boot.model.naming.PhysicalNamingStrategyStandardImpl
```
Comment (MrSebastian) [inline praktikumsplaner-backend/src/main/resources/application-dev.yml:22]: are u sure we can delete the Naming-Strategy? I had some trouble with `dueDate` in bei TODO-App. Without the strategy, hibernate was looking for a column `due_date` but my column was `dueDate`.

File: praktikumsplaner-backend/src/main/resources/application-dev.yml [hunk 3]
```diff
@@ -13,13 +13,7 @@ spring:
   h2.console.enabled: true
   jpa:
     database: H2
-    hibernate:
-        # always drop and create the db should be the best
-        # configuration for local (development) mode. this
-        # is also the default, that spring offers by convention.
-        # but here explicite:
-        ddl-auto: create-drop
```
Comment (MrSebastian) [inline praktikumsplaner-backend/src/main/resources/application-dev.yml:21]: i would change this to `validate` and move it in the default profile (`application.yml`).

File: praktikumsplaner-backend/src/main/resources/application-dev.yml (hunk 2 above)
Comment (Tuckerle) [reply praktikumsplaner-backend/src/main/resources/application-dev.yml:22]: Moved to application.yml

**Review by LuisEinig — APPROVED** [review]

**Pull request merged**